src/classes/ADDR_SmartyStreets_Validator.cls

```diff
@@ -143,69 +143,70 @@ public with sharing class ADDR_SmartyStreets_Validator implements ADDR_IValidato
-	                    av.Verification_Status__c = Label.Addr_Verified;
-	                }
-	            }
-	            for (Address__c av : avList) {             
-	                if (av.API_Response__c == null && !String.isBlank(settings.Auth_ID__c) && 
-	                !String.isBlank(settings.Auth_Token__c) && (settings.Class__c == 'ADDR_SmartyStreets_Validator')) {   
-	                    av.API_Response__c = Label.Address_Not_Found;
-	                    av.Verification_Status__c = Label.Address_Not_Found;
-	                }
-	            }
-	        } catch (Exception ex) {
-	            for (Address__c av : avList) {
-	                UTIL_Debug.debug('****Flagging address as NOT verified Address ID: ' + av.Id 
-	                               + ' - Exception in verifyRecords100: ' + ex.getMessage() 
-	                               + '\n' + ex.getStackTraceString());
-	                av.Verified__c = false;
-	                av.API_Response__c = ex.getMessage();
-	                av.Verification_Status__c = ex.getMessage();
-	                ERR_Handler.processError(ex, ERR_Handler_API.Context.ADDR);
-	            }
-	        }
+                            av.MailingStreet__c = sr.delivery_line_1;
+                            av.MailingStreet2__c = sr.delivery_line_2;
+                            av.MailingCity__c = sr.components.city_name;
+                            av.MailingState__c = sr.components.state_abbreviation;
+                            av.MailingPostalCode__c = (sr.components.zipcode == null) ? null : (sr.components.zipcode +
+                                ((sr.components.plus4_code != null && sr.components.plus4_code != '') ? ('-' + sr.components.plus4_code) : ''));
+                            av.MailingCountry__c = 'US';
+                            av.Geolocation__Latitude__s = sr.metadata.latitude;
+                            av.Geolocation__Longitude__s = sr.metadata.longitude;
+                            av.County_Name__c = sr.metadata.county_name;
+                            av.Congressional_District__c = sr.metadata.congressional_district;
+                        } else {
+                            // more than one candidate means should mark as ambiguous
+                            UTIL_Debug.debug('****Flagging address as NOT verified. Flagging it as ambiguous. Address ID: ' + av.Id);
+                            av.Verified__c = true;
+                            av.Ambiguous__c = true;
+                            responseGroup.add(sr);
+                        }
+                        String response = JSON.serializePretty(responseGroup);
+                        av.API_Response__c = trimResponse(response);
+                        av.Verification_Status__c = Label.Addr_Verified;
+                    }
+                }
+                for (Address__c av : avList) {             
+                    if (av.API_Response__c == null && !String.isBlank(settings.Auth_ID__c) && 
```

Comment:
Is this check on the Auth ID, Auth Token and Class a little late? Should we be doing this check at an earlier point in the code? If not, why is this required here, but not in verifyZipRecords?

Reply:
Agreed that this is a really weird place for this check. I'll have to dig in a bit to figure out if we should move it up or remove it altogether.

Reply:
Yep, we can remove these checks since we'll only get to this code path if the class setting is set to this class, and we're already checking if the auth vars exist on the ADDR_SmartyStreets_Gateway lines [54](https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/ADDR_SmartyStreets_Gateway.cls#L54) and [185](https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/ADDR_SmartyStreets_Gateway.cls#L185). I'll remove it.